addons/web-support/impl/src/main/java/org/jboss/windup/web/addons/websupport/rest/graph/EJBResourceImpl.java

```diff
@@ -0,0 +1,86 @@
+package org.jboss.windup.web.addons.websupport.rest.graph;
+
+import java.util.*;
+
+import org.jboss.windup.graph.GraphContext;
+import org.jboss.windup.graph.model.BelongsToProject;
+import org.jboss.windup.graph.model.ProjectModel;
+import org.jboss.windup.graph.service.GraphService;
+import org.jboss.windup.rules.apps.javaee.model.*;
+import org.jboss.windup.web.addons.websupport.model.ReportFilterDTO;
+
+public class EJBResourceImpl extends AbstractGraphResource implements EJBResource
+{
+    @Override
+    public Object getMDB(Long reportID, Map<String, Object> filterAsMap)
+    {
+        return this.getFilteredData(reportID, null, filterAsMap, EjbMessageDrivenModel.class);
+    }
+
+    @Override
+    public Object getEJB(Long reportID, String sessionType, Map<String, Object> filterAsMap)
+    {
+        return this.getFilteredData(reportID, sessionType,  filterAsMap, EjbSessionBeanModel.class);
+    }
+
+    @Override
+    public Object getEntity(Long reportID, Map<String, Object> filterAsMap)
+    {
+        return this.getFilteredData(reportID, null, filterAsMap, EjbEntityBeanModel.class);
+    }
+
+    private <T extends EjbBeanBaseModel> Object getFilteredData(Long reportID, String sessionType, Map<String, Object> filterAsMap, Class<T> clazz) {
+        GraphContext graphContext = this.getGraph(reportID);
+
+        ReportFilterDTO filter = this.reportFilterService.getReportFilterFromMap(filterAsMap);
+        Set<String> includeTags = new HashSet<>();
+        Set<String> excludeTags = new HashSet<>();
+        Set<ProjectModel> projectModels = null;
+
+        if (filter.isEnabled())
+        {
+            includeTags.addAll(filter.getIncludeTags());
+            excludeTags.addAll(filter.getExcludeTags());
+            projectModels = this.getProjectModels(graphContext, filter);
+        }
+
+        List<T> ejbMessageDrivenArrayList = new ArrayList<>();
+        GraphService<T> ejbMessageDrivenModelService = new GraphService<>(graphContext, clazz);
+        Iterable<T> data;
+        if (sessionType != null)
+        {
+            data = ejbMessageDrivenModelService.findAllByProperties(new String[]{"sessionType"}, new String[]{sessionType});
```

Comment:
I think that I had said earlier that this line was problematic, but I was mistaken. In looking at the GraphService code again, the underlying query doesn't work the way that I had remembered it. Good approach. :) My only suggestion would be to use org.jboss.windup.rules.apps.javaee.model.EjbBeanBaseModel#SESSION_TYPE instead of a literal string.